R/utilities-simulation-results.R

```diff
@@ -0,0 +1,132 @@
+#' Get simulated values of molecules or parameters.
+#'
+#' Return values for given entities generated in a individual simulation.
+#'
+#' The function receives an object of simulation results generated by running the simulation
+#' and returns time-values profiles for the chosen quantities. If the results contain more then one individual,
+#' an error is thrown. In this case, \code{getPopulationResultsValues} should be used.
+#'
+#' @inheritParams getPopulationResultsValues
+#'
+#' @return A list of results, each entry having the name of a quantity path and consisting of the element \code{x} being the time values (in minutes)
+#' and the element \code{y} a \code{Nx1} array of simulated values, with \code{N} being the number of entries (equals the number of output time steps).
+#' If no results were generated for a given path, the value of the element is \code{NULL}.
+#'
+#' @seealso \code{\link{getPopulationResultsValues}}
+#'
+#' @export
+#'
+#' @examples
+#' simPath <- system.file("extdata", "simple.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simPath)
+#' simResults <- runSimulation(sim)
+#'
+#' paths <- c(
+#'   "Organism|PeripheralVenousBlood|Caffeine|Plasma (Peripheral Venous Blood)",
+#'   "Organism|ArterialBlood|Plasma|Caffeine|Concentration in container"
+#' )
+#'
+#' resultsValues <- getResultsValues(simResults, paths)
+#' # plot(resultsValues$`Organism|PeripheralVenousBlood|Caffeine|Plasma (Peripheral Venous Blood)`)
+getResultsValues <- function(simulationResults, quantitiesOrPaths) {
```

Comment:
`getResultsValues` is a helper function that calls `getPopulationResultsValues`.

Reply:
I don't understand this methods. IT calls `getPopulationResultsValues` with exactly the same arguments. Why do we need this at all?

Reply:
aside from the check it is exactly the same method

Reply:
Yes - it's just for convenience to differentiate between population and individual simulations.

Reply:
Still. Why do we ned to method. Same input, and same outputs. Aren't those methods equivalent?

Looking at the code, I would rename getPopulationResultsValues to getResultsValues and delete getResultsValues 

Or am I missing something ?

Reply:
The only difference is that `getResultsValues` stops if trying to receive results with more than one individual - safety check that we are not using a population simulation where we expect a single individual...

But we can leave only one of the methods if there is no demand for this check.

Reply:
@PavelBal I would not know what would be the benefit for now. The beauty of having the same structure for single and multiple results (ind vs pop sim) is that you don't have to call different methods depending on the use case.